pyomo/contrib/doe/measurements.py

```diff
@@ -61,216 +63,74 @@ def __init__(self, measurement_index_time, variance=None, ind_string='_index_'):
-                for ind in measure_name_and_index[j]:
-                    flatten_name = j + self.ind_string + str(ind)
-                    flatten_names.append(flatten_name)
-            else:
-                flatten_names.append(j)
-
-        self.flatten_measure_name = flatten_names
-
-    def _generate_variance(self, flatten_measure_name, variance, name_and_index):
-        """
-        Generate the variance dictionary
-        Parameters
-        ----------
-        flatten_measure_name: flattened measurement names. For e.g., flattenning {'C':{'CA': 10, 'CB': 1, 'CC': 2}} will be 'C_index_CA', ..., 'C_index_CC'.
-        variance:
-            a ``dict``, keys are measurement variable names, values are a dictionary, keys are its extra index name,
-            values are its variance as a scalar number.
-            For e.g., for the kinetics illustrative example, it should be {'C':{'CA': 10, 'CB': 1, 'CC': 2}}.
-            If given None, the default is {'C':{'CA': 1, 'CB': 1, 'CC': 1}}.
-            If there is no extra index, it is a dict, keys are measurement variable names, values are its variance as a scalar number.
-        name_and_index:
-            a dictionary, keys are measurement names, values are a list of extra indexes.
-        """
-        flatten_variance = {}
-        for i in flatten_measure_name:
-            if variance is None:
-                flatten_variance[i] = 1
-            else:
-                # split the flattened name if needed
-                if self.ind_string in i:
-                    measure_name = i.split(self.ind_string)[0]
-                    measure_index = i.split(self.ind_string)[1]
-                    if type(name_and_index[measure_name][0]) is int:
-                        measure_index = int(measure_index)
-                    flatten_variance[i] = variance[measure_name][measure_index]
-                else:
-                    flatten_variance[i] = variance[i]
-        self.flatten_variance = flatten_variance
-
-    def _generate_flatten_timeset(self, all_info, flatten_measure_name,name_and_index):
-        """
-        Generate flatten variables timeset. Return a dict where keys are the flattened variable names,
-        values are a list of measurement time.
-
-        """
-        flatten_measure_timeset = {}
-        for i in flatten_measure_name:
-            # split the flattened name if needed
-            if self.ind_string in i:
-                measure_name = i.split(self.ind_string)[0]
-                measure_index = i.split(self.ind_string)[1]
-                if type(name_and_index[measure_name][0]) is int:
-                    measure_index = int(measure_index)
-                flatten_measure_timeset[i] = all_info[measure_name][measure_index]
-            else:
-                flatten_measure_timeset[i] = all_info[i]
-        self.flatten_measure_timeset = flatten_measure_timeset
-
-    def _model_measure_name(self):
+        if self_define_res:
```

Comment:
This is not the interface we discussed. We worked out having an `add_measurement` function that the user would call for each measurement and specify indices. This function would then build up the data stored in `self.measurement_name`.